src/internal/got/client.ts

```diff
@@ -684,14 +686,41 @@ export class GotCFToolsClient implements CFToolsClient {
             return playerId;
         }
 
-        const response = await this.client.get<GetUserLookupResponse>('v1/users/lookup', {
-            searchParams: {
-                identifier: playerId.id,
-            },
-            context: {
-                authorization: await this.auth!.provide(this.client),
-            },
-        });
+        const requestUsesAccountCreation = this.hasAccountCreationAccess === true
+            && this.enterpriseToken !== undefined
+            && playerId instanceof SteamId64;
+
+        let response;
+        try {
+            response = await this.client.get<GetUserLookupResponse>('v1/users/lookup', {
+                searchParams: {
+                    identifier: playerId.id,
+                },
+                context: {
+                    authorization: await this.auth!.provide(this.client),
+                },
+            });
+        } catch (err) {
+            // Note: We could just apply the `create` param above, but this parameter is only ever
+            // allowed when there is no existing user with the supplied identity token.
+            if (requestUsesAccountCreation) {
```

Comment:
This should only be tried, if the CFTools response is a 404 Not Found. If the error indicates another issue, we should not make a secondary request, which might fail as well.

Reply:
I agree, but as stated in my PR description - I'm not sure about the details/implementation regarding this:

> I would like to request you to look further into the [error handler](https://github.com/FlorianSW/cftools-sdk/pull/15/files#diff-dc3bab5e8f385c8749856e8d81bdfb65df2103c0cd2d73f14323a324222dbe22R704-R724), as we should be checking if the error is a ResourceNotFound error, and I'm also not sure this catch logic should be in here, or if it should be handled in your withErrorHandler wrapper.

There's no isResourceNotFoundError utility (type predicate) for example, and I'm not sure if this logic should be here, or in your error handler wrapper.

I would appreciate it if you could ammend a quick commit to this PR for this specifically 🙏

Reply:
From your initial comment:

> based on the `autoCreateAccount` property, add the create URL parameter to the resolve request or not.

This would (technically) remove the need for internal error handling for this, but the library uses the #resolve function internally, and we need a way to globally enable this `create` retry functionality for these internal requests. Otherwise users of the sdk have to look into the source code, determine where the #resolve function is used internally, and add external logic to handle/retry with the parameter, followed by running the same initial function again. This is outside of the internal function logic, and would be a huge headache that isn't technically needed.

This is why the `withAccountCreation` builder method could be beneficial, as we need a way to tell the SDK to retry #resolved internally with the `create` parameter globally.

The following endpoints/function all use this resolve function internally, and would need a global override for retry logic with the `create` paramater:

- (get/delete)PlayerDetails
- (get/put/delete)PriorityQueue
- (get/put/delete)Whitelist
- listBans
- putBan

Reply:
I have added dynamic support for this in https://github.com/FlorianSW/cftools-sdk/pull/15/commits/527d47b4d0aa95fe45b29a0c03a10db24af19924. Feel free to exclude this commit from the PR if you're thinking something else, or I can amment a revert commit to this PR instead - let me know.

Reply:
Oh my 🙈 

> philipp — Today at 5:43 PM
> If the app does not have permission to use it:
> 403 with error field set to a message
> 
> If the account creation was called with anything other then a steam64
> 400 with error field set to a message
> 
> If the steam64 is invalid, either wrong universe (perfect world) or steam debug or steam testing
> 400 with error field set to a message
> 
> If the account creation itself fails
> 500 with no explicit message
> 
> If everything goes through
> 200 with notice field set to Account Creation API account created
> The rest will be formatted as usual, eg status=true and cftools_id

Reply:
> This would (technically) remove the need for internal error handling for this, as it could be handled externally with this option, but the library uses the #resolve function internally, and we need a way to globally enable this create retry functionality for these internal requests. Otherwise users of the sdk have to look into the source code, determine where the #resolve function is used internally, and add external logic to handle/retry with the parameter, followed by running the same initial function again. This is outside of the internal function logic, and would be a huge headache that isn't technically needed.

Good point, didn't think about this 🙈 



> Oh my 🙈
> 
> > philipp — Today at 5:43 PM
> > If the app does not have permission to use it:
> > 403 with error field set to a message
> > If the account creation was called with anything other then a steam64
> > 400 with error field set to a message
> > If the steam64 is invalid, either wrong universe (perfect world) or steam debug or steam testing
> > 400 with error field set to a message
> > If the account creation itself fails
> > 500 with no explicit message
> > If everything goes through
> > 200 with notice field set to Account Creation API account created
> > The rest will be formatted as usual, eg status=true and cftools_id

I mean, that's fine :)